src/main/java/com/worksap/fig/lang/Seq.java

```diff
@@ -404,4 +431,189 @@ default int findLastIndex(Predicate<T> condition) {
     default Seq<T> subSeq(int fromIndex, int toIndex) {
         return new SeqImpl<>(this.subList(fromIndex, toIndex));
     }
+
+    /**
+     * Remove elements which satisfy the condition, resulting a new seq without changing the original one.
+     *
+     * @return The new seq after the change
+     */
+    default Seq<T> removeIf(BiPredicate<T, Integer> condition) {
```

Comment:
To keep consistency with other method styles, I think we need a lambda for only the element without index?